### DISCUSSION TITLE
Different formatting options for Mac Addresses.

### DISCUSSION THREAD

**Discussion by AdamMason1877** [Cacti/cacti#5746] [Ideas]
Currently in order to search our network cores, Cisco based, for the mac addresses under mac tracker, we have to change the mac address format to xxxx.xxxx.xxxx manually because it is not output as such in Mac Track exports. 

Currently to work around this I use Excel formula, 

`=SUBSTITUTE(MID(F2,1,2)&MID(F2,4,2),"-", ".") & "." & SUBSTITUTE(MID(F2,7,2)&MID(F2,10,2),"-", ".") & "." & SUBSTITUTE(MID(F2,13,2)&MID(F2,16,2),"-", ".")`

F2 in this case being the cell name of the original mac value. Is there a location in settings that specifies mac formatting? If not, where is the code, I would like to push this option to the repository.

**Comment by xmacan** [comment]
Mactrack has only "Mac Address Delimiter" in settings. If you want to extend setting for mac address format and display mac address in configured format, your pull request is welcome. I suggest to preserve current format in database, use custom format only for view, email and reports.

**Comment by TheWitness** [comment]
Why not just add another option here?

![image](https://github.com/Cacti/cacti/assets/1439914/1a372334-5ed7-4491-bdcd-40f98360715a)

**Comment by xmacan** [comment]
I will do it. Closing here, opening feature request in mactrack

**Comment by xmacan** [comment]
@AdamMason1877 

fixed here https://github.com/Cacti/plugin_mactrack/pull/253
I suggest create mysqldump of mactrack tables for sure. I'm changing format in db.
